drv/gimlet-seq-server/src/main.rs

```diff
@@ -746,10 +752,17 @@ impl<S: SpiServer> ServerImpl<S> {
                     self.seq
                         .read_bytes(Addr::A1SMSTATUS, &mut status)
                         .unwrap_lite();
-                    ringbuf_entry!(Trace::A1Status(status[0]));
 
-                    if status[0] == Reg::A1SMSTATUS::A1SmEncoded::Done as u8 {
-                        break;
+                    let a1sm = A1SmStatus::try_from(status[0]);
+                    ringbuf_entry!(Trace::A1Status(a1sm));
+
+                    match a1sm.ok() {
+                        Some(state) => {
+                            if state == A1SmStatus::Done {
+                                break;
+                            }
+                        }
+                        None => (),
```

Comment:
```suggestion
                    if a1sm == Ok(A1SmStatus::Done) {
                        break;
```
(and same trick can be applied below)